tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -243,16 +244,44 @@ struct UnsortedSegmentSumFunctor<CPUDevice, T, Index> {
     }
   }
 };
-
+// UnsortedSegmentMaxFunctor implementation for CPUDevice.
+template <typename T, typename Index>
+struct UnsortedSegmentMaxFunctor<CPUDevice, T, Index>
+    : UnsortedSegmentBaseFunctor<CPUDevice, T, Index> {
+  void operator()(OpKernelContext* ctx, const CPUDevice& d,
+                  const Index output_rows, const TensorShape& segment_ids_shape,
+                  typename TTypes<Index>::ConstFlat segment_ids,
+                  const Index data_size, const T* data,
+                  typename TTypes<T, 2>::Tensor output) override {
+    output.setConstant(std::numeric_limits<T>::min());
```

Comment:
This doesn't share any code with UnsortedSegmentSumFunctor, yet there is lots of code copied between the two (I think the only differences are the initializer on line 256 and the output update on line 268).

Is it possible to have the base UnsortedSegmentBaseFunctor contain this shared code, and pass in "initializer" and "reducer" functors into the base implementation which can be called at these two call sites?